### DISCUSSION TITLE
Cannot use cookies.set in SvelteKit 2 because of hooks. OAuth2

### DISCUSSION THREAD

**Discussion by jramiroz98** [pocketbase/pocketbase#4050] [Q&A] [answered]
Currently I'm trying to store the state and verifier from google auth in a cookie. I'm doing manual implementation for it. The problem lies in that, using the following line of code in my hooks file, I can't use the `cookies.set()` in my  form action.

This line seems to block my cookie creation process in hooks:
`response.headers.set('set-cookie', event.locals.pb.authStore.exportToCookie({ secure: false }))`

By commenting the previous line I am able to set cookies. I don't want to just erase it as I'm afraid it might break something with pocketbase or even worse, open a vulnerability.

What am I doing wrong and how can I fix this?

Thanks in advance! Have an awesome day! ☺️

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I fully understand your description, but if your issue is because `response.headers.set` overwrites your cookies added via `cookies.set`, then you could try:

- to use [`response.headers.append()`](https://developer.mozilla.org/en-US/docs/Web/API/Headers/append)
- OR replace `response.headers.*` and the `exportToCookie` helper entirely and instead use the SvelteKit cookies abstraction, something like:
  ```js
  import PocketBase, { getTokenPayload } from "pocketbase";
 
  cookies.set(
    "pb_auth",
    JSON.stringify({token: event.locals.pb.authStore.token}),
    {
      "secure": true,
      "httpOnly": true,
      "sameSie": "strict",
      "path": "/",
      "expires": new Date(getTokenPayload(event.locals.pb.authStore.token).exp * 1000)
    }
  )
  ```

**Reply by ganigeorgiev** [reply]
>  don't want to just erase it as I'm afraid it might break something with pocketbase or even worse, open a vulnerability.

Note that PocketBase doesn't use cookies and you can't introduce vulnerabilities in PocketBase just by setting a cookie value.
The cookie is just a convenient token share mechanism between the user browser and your node server.

**Comment by jramiroz98** [comment]
Used [response.headers.append()](https://developer.mozilla.org/en-US/docs/Web/API/Headers/append) and worked like a charm! Thank you so much! Have an awesome day!

**Reply by MuslemRahimi** [reply]
Worked for me too. Thanks for asking this question!
